### DISCUSSION TITLE
Simple walker class which outputs XML

### DISCUSSION THREAD

**Discussion by awsteiner** [zeux/pugixml#480] [Show and tell]
I created a simple walker class (ostream_walker) which outputs the XML (e.g. to std::cout) starting with a pugi::xml_node object at https://github.com/awsteiner/o2scl/blob/3962b8e9ee69a634ac0457bae21017efd8c56e62/src/other/xml.h . Let me know if it's useful to someone, e.g. if you want to use it for the documentation. I'm sure it's not perfect, but it may be helpful for others.

**Comment by zeux** [comment]
Is this substantially different from calling `xml_node::print` with the stream as the argument?

**Comment by awsteiner** [comment]
It's simpler, i.e. it doesn't include attribute information. I shared it not because the functionality was particularly interesting, but rather because seeing a slightly more complex walker might be helpful for users who would like to design their own walkers.

**Comment by awsteiner** [comment]
In either case thank you for your excellent work.

**Comment by ghost** [comment]
> I shared it not because the functionality was particularly interesting

It should not be an issue, but a feature? Or moving it to [Discussions](https://github.com/zeux/pugixml/discussions) may be a better fit?

**Comment by zeux** [comment]
I'm going to convert to a discussion since the documentation already has a simple example of a walker implementation.
